### PR TITLE
Fix Android Variants

### LINKED ISSUE #30: Module-based artifact whose Android variant was selected cannot resolve pom for common artifact
Repro incoming

### PR DESCRIPTION
**Pull request by hfhbd**
This fixes android variants, published with modules file and artifacts published without modules file eg stdlib-js

Fixes #30

### CODE REVIEW

**Comment by hfhbd** [comment]
@JakeWharton What do you need to merge it? More js/mpp tests?

**Comment by JakeWharton** [comment]
I haven't looked yet.

File: src/main/kotlin/app/cash/licensee/dependencyGraph.kt [hunk 1]
```diff
@@ -152,27 +157,68 @@ private fun loadDependencyCoordinates(
   }
 }
 
+private fun Project.pomConfiguration(
+  pomDependency: Dependency,
+  variants: List<ResolvedVariantResult>
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/dependencyGraph.kt]:
```suggestion
  variants: List<ResolvedVariantResult>,
```

File: src/main/kotlin/app/cash/licensee/dependencyGraph.kt [hunk 2]
```diff
@@ -152,27 +157,68 @@ private fun loadDependencyCoordinates(
...
+  .lenientConfiguration
+
+private val LenientConfiguration.resolvedFiles get() = allModuleDependencies
+  .flatMap { it.allModuleArtifacts }
+  .map { it.file }
+
+private fun getPomFile(
+  project: Project,
+  pomCoordinates: String,
+  variants: List<ResolvedVariantResult>
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/dependencyGraph.kt]:
```suggestion
  variants: List<ResolvedVariantResult>,
```

File: src/main/kotlin/app/cash/licensee/dependencyGraph.kt [hunk 3]
```diff
@@ -152,27 +157,68 @@ private fun loadDependencyCoordinates(
...
-    .lenientConfiguration
-
-  val resolvedFiles = pomConfiguration
-    .allModuleDependencies
-    .flatMap { it.allModuleArtifacts }
-    .map { it.file }
+  val (pomConfiguration, resolvedFiles) = getPomFile(
+    project = project,
+    pomCoordinates = pomCoordinates,
+    variants = variants
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/dependencyGraph.kt]:
```suggestion
    variants = variants,
```

File: src/main/kotlin/app/cash/licensee/dependencyGraph.kt [hunk 4]
```diff
@@ -152,27 +157,68 @@ private fun loadDependencyCoordinates(
...
+    variants = emptyList()
+  )
+  val resolvedFilesWithoutVariants = pomConfigurationWithoutVariants.resolvedFiles
+
+  return if (resolvedFilesWithoutVariants.isNotEmpty()) {
+    pomConfigurationWithoutVariants to resolvedFilesWithoutVariants
+  } else {
+    val pomConfigurationWithVariants = project.pomConfiguration(
+      pomDependency = pomDependency,
+      variants = variants
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/dependencyGraph.kt]:
```suggestion
      variants = variants,
```

File: src/main/kotlin/app/cash/licensee/dependencyGraph.kt [hunk 5]
```diff
@@ -152,27 +157,68 @@ private fun loadDependencyCoordinates(
...
+private fun getPomFile(
+  project: Project,
+  pomCoordinates: String,
+  variants: List<ResolvedVariantResult>
+): Pair<LenientConfiguration, List<File>> {
+  val pomDependency = project.dependencies.create(pomCoordinates)
+
+  val pomConfigurationWithoutVariants = project.pomConfiguration(
+    pomDependency = pomDependency,
+    variants = emptyList()
```
Comment (JakeWharton) [inline src/main/kotlin/app/cash/licensee/dependencyGraph.kt]:
```suggestion
    variants = emptyList(),
```

**Review by JakeWharton — APPROVED** [review]

**Pull request merged**